sparrow-blog-main/src/main/java/com/sparrow/blog/config/MvcConfigurerAdapter.java

```diff
@@ -0,0 +1,46 @@
+package com.sparrow.blog.config;
+
+import com.sparrow.protocol.ClientInformation;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.MethodParameter;
+import org.springframework.web.bind.support.WebDataBinderFactory;
+import org.springframework.web.context.request.NativeWebRequest;
+import org.springframework.web.method.support.HandlerMethodArgumentResolver;
+import org.springframework.web.method.support.ModelAndViewContainer;
+import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;
+
+import javax.servlet.http.HttpServletRequest;
+import java.util.List;
+
+@Configuration
+public class MvcConfigurerAdapter implements WebMvcConfigurer {
+    /**
+     * 配置静态访问资源
+     * <p>
+     * http://localhost:8000/s/static.html
+     *
+     * @param registry
+     */
+    @Override
+    public void addResourceHandlers(ResourceHandlerRegistry registry) {
+        registry.addResourceHandler("/s/**").addResourceLocations("classpath:/static/");
+    }
+
+    @Override public void addArgumentResolvers(List<HandlerMethodArgumentResolver> argumentResolvers) {
+        argumentResolvers.add(new HandlerMethodArgumentResolver() {
+            @Override public boolean supportsParameter(MethodParameter parameter) {
+                return parameter.getParameter().getParameterizedType().equals(ClientInformation.class);
+            }
+
+            @Override public Object resolveArgument(MethodParameter parameter, ModelAndViewContainer container,
+                NativeWebRequest request, WebDataBinderFactory factory) throws Exception {
+                HttpServletRequest req = (HttpServletRequest) request.getNativeRequest();
+                ClientInformation client = new ClientInformation();
+                //todo 解析 client 自定义参数
+                client.setIp("127.0.0.1");
+                return client;
```

Comment:
I don't know why you get the `NativeRequets` but do not use or return them.

Reply:
And could you please extract the `127.0.0.1` to the static value or move it into properties?

Reply:
@sparrowzoo There is an unknown code, could you please help to explain why the request is not used and not rerun?